core/embed/rust/src/ui/model_tr/component/homescreen.rs

```diff
@@ -148,24 +169,64 @@ where
 
     fn place(&mut self, bounds: Rect) -> Rect {
         self.label.place(LABEL_AREA);
+        self.loader.place(AREA);
         bounds
     }
 
     fn event(&mut self, ctx: &mut EventCtx, event: Event) -> Option<Self::Msg> {
         Self::event_usb(self, ctx, event);
-        // HTC press of any button will lock the device
-        if let Some(ButtonControllerMsg::Triggered(..)) = self.invisible_buttons.event(ctx, event) {
-            return Some(());
+
+        // Only care about button and loader events when there is a possibility of
+        // locking the device
+        if let Some(self_loader) = &mut self.loader {
+            // When loader has completely grown, we can lock the device
+            if let Some(LoaderMsg::GrownCompletely) = self_loader.event(ctx, event) {
+                return Some(());
+            }
+
+            // Longer hold of any button will lock the device.
+            // Normal/quick presses and releases will show/hide the loader.
+            let button_event = self.invisible_buttons.event(ctx, event);
+            if let Some(ButtonControllerMsg::Pressed(..)) = button_event {
+                if !self.show_loader {
+                    self.show_loader = true;
+                    self_loader.mutate(ctx, |ctx, loader| {
+                        loader.start(ctx);
+                        ctx.request_paint();
+                    });
+                }
+            }
+            if let Some(ButtonControllerMsg::Triggered(..)) = button_event {
+                self.show_loader = false;
+                self_loader.mutate(ctx, |ctx, loader| {
+                    loader.stop(ctx);
+                    ctx.request_paint();
+                });
+            }
         }
+
         None
     }
 
     fn paint(&mut self) {
-        // Painting the homescreen image first, as the notification and label
-        // should be "on top of it"
-        self.paint_homescreen_image();
-        self.paint_notification();
-        self.paint_label();
+        // Redraw the whole screen when the screen changes (loader vs homescreen)
+        if self.show_loader {
+            if !matches!(self.current_screen, CurrentScreen::Loader) {
+                rect_fill(AREA, theme::BG);
```

Comment:
Let's use `display::clear()` instead, it might be faster (not sure if it actually is on T2B1) than drawing generic rectangle.

Reply:
Done in the force-push